checkfield/checkfield.go

```diff
@@ -18,15 +19,11 @@ func CheckRequiredFields(msg interface{}, requiredFields []string) error {
 			switch f.Kind() {
 			case reflect.String:
 				if f.String() == "" {
-					return status.Errorf(codes.InvalidArgument, "Required field %s is not provided", fieldName)
+					return status.Errorf(codes.InvalidArgument, "required field `%s` is not provided", fieldName)
 				}
 			case reflect.Ptr:
 				if f.IsNil() {
-					return status.Errorf(codes.InvalidArgument, "Required field %s is not provided", fieldName)
-				} else if reflect.Indirect(reflect.ValueOf(f)).Kind() == reflect.Struct {
-					if err := CheckRequiredFields(f.Interface(), requiredFields); err != nil {
```

Comment:
The `requiredFields` is for struct msg, not for any field in msg, so it doesn't make sense to use recursion here.

Reply:
It needs to be recursive in a nested message case. The protobuf message will be converted into a struct type in Go. Add a nested struct test case?

Reply:
Since recursive checking is required for certain proto message, I will bring it back.